lib/async_hooks.js

```diff
@@ -77,13 +77,13 @@ function fatalError(e) {
 
 class AsyncHook {
   constructor({ init, before, after, destroy }) {
-    if (init && typeof init !== 'function')
+    if (typeof init !== 'undefined' && typeof init !== 'function')
```

Comment:
I see. Can we at least use `init !== undefined` instead, that is the form we use elsewhere in core?